### ISSUE TITLE
IPFS resource on non-IPFS page tries to redirect to https://cid.ipfs.localhost:8080

### ISSUE THREAD

**Issue by Jontes-Tech** [ipfs/ipfs-companion#1324]
**Describe the bug**
In chromium, http://localhost:3000/ipfs/cid redirects to https://cid.ipfs.localhost:8080. Kubo's gateway only serves HTTP, not HTTPS.

**To Reproduce**
1. Create an HTTP file.
2. Create an <img> pointing to https://ipfs.io/ipfs/any-cid-here
3. Image will fail to load because it tries to redirect to https on 8080.

**Expected behavior**
Image loads successfully

**Screenshots**
![Screenshot from 2024-06-08 15-17-04](https://github.com/ipfs/ipfs-companion/assets/84264740/8b3b66de-6197-4114-9cc4-7c028d54665e)

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Browser: Chromium and Google's "Chrome" rebrand
 - Version: Latest versions on Chrome Web Store.

**Comment by lidel** [comment]
@Jontes-Tech I was unable to reproduce in fresh install on Chrome 126.0.6478.36

- Opening http://localhost:3000/ipfs/cid does not redirect to https://cid.ipfs.localhost:8080.
- Bit confused what steps (1) and (2) mean – mind rephrasing? Looks like you've cut some important text.

**Comment by Jontes-Tech** [comment]
Yeah, sorry. Github stripped away my HTML tags. I've edited the message to make it clearer. 

Basically http://localhost:8080/ipfs/cid should redirect to http://cid.ipfs.localhost:8080 but instead redirects to https://cid.ipfs.localhost:8080 when served as an image in an HTTPS context.

One example of this is present on the following webpage: https://ens.page/jontes.eth

You will see in the network tab how IPFS companion redirects to https:// on localhost.

See the screenshots I've attached:
![image](https://github.com/ipfs/ipfs-companion/assets/84264740/937e3aab-7f79-4308-90b0-1865d76fd2b1)
![image](https://github.com/ipfs/ipfs-companion/assets/84264740/9af490fb-e172-4715-90fc-6013776c9af2)


Reproducible on both Chrome and Chromium, 125.0.6422.141 (Official Build) Arch Linux (64-bit)
